### PR TITLE
Add Dockerfiles for 5.3 nightlies on the new platforms

### PR DESCRIPTION
**Pull request by gwynne**
Adds new `Dockerfile`s for building nightlies on the newly supported Linux platforms for which nightlies of the 5.3 snapshots do not already exist:

##### `swift-5.3-branch`

- Ubuntu 20.04 "Focal" (`swiftlang/swift:nightly-5.3-focal`)
- Ubuntu 20.04 "Focal", slim (`swiftlang/swift:nightly-5.3-focal-slim`)
- CentOS 7 (`swiftlang/swift:nightly-5.3-centos7`)
- CentOS 7, slim (`swiftlang/swift:nightly-5.3-centos7-slim`)
- CentOS 8 (`swiftlang/swift:nightly-5.3-centos8`)
- CentOS 8, slim (`swiftlang/swift:nightly-5.3-centos8-slim`)
- AmazonLinux 2 (`swiftlang/swift:nightly-5.3-amazonlinux2`)
- AmazonLinux 2, slim (`swiftlang/swift:nightly-5.3-amazonlinux2-slim`)

### CODE REVIEW

**Comment by shahmishal** [comment]
@gwynne Thanks for creating the pull request! 

We currently don't have 5.3 branch packages for Ubuntu 20.04, CentOS 8, and Amazon Linux 2. I am working on getting this configured but it's not there yet. 

The reason I have not created nightly 5.2 images for Ubuntu 20.04, CentOS 8, and Amazon Linux 2 because currently we are in dot release mode. Normally, we only pick up few fixes and nightly images are built 2-3 time a dot release. 

We can merge this PR, and once we start having nightly toolchains it will be useful.

**Review by shahmishal — APPROVED** [review]

**Comment by gwynne** [comment]
@swift-ci please test

**Comment by gwynne** [comment]
@shahmishal CI fails due to there being no `latest-build.yml` for `swift-5.[2|3]-branch/ubuntu2004`, of course, but as far as I'm aware that should be the only blocker.

**Comment by 1oo7** [comment]
You're missing CentOS 7!!!

**Comment by shahmishal** [comment]
> You're missing CentOS 7!!!

CentOS 7 is not currently available on swift.org, once it becomes available we can add it.

**Comment by 1oo7** [comment]
I noticed there is a Centos7 Dockerfile for Swift CI now, so just curious what next steps are needed to get a Swift that's been built on Centos7?

**Comment by gwynne** [comment]
@shahmishal Any motion here yet?

File: nightly-5.3/centos/8/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,70 @@
+FROM centos:8
+LABEL maintainer="Swift Infrastructure <swift-infrastructure@swift.org>"
+LABEL description="Docker Container for the Swift programming language"
+
+RUN yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
+
+RUN yum -y update && yum install --enablerepo=PowerTools -y \
```
Comment (shahmishal) [inline nightly-5.3/centos/8/Dockerfile]:
```suggestion
RUN yum install --enablerepo=PowerTools -y \
```
Can you drop `yum -y update` from all Dockerfiles?

**Comment by shahmishal** [comment]
Swift 5.3 development snapshots available -- https://swift.org/download/#snapshots
* Ubuntu 16.04
* Ubuntu 18.04
* CentOS 7
* CentOS 8
* Amazon Linux 2 

Ubuntu 20.04 coming soon.

File: nightly-5.3/centos/8/Dockerfile (hunk 1 above)
Comment (gwynne) [reply nightly-5.3/centos/8/Dockerfile]: @shahmishal Branch updated and `yum -y update` removed from all.

**Comment by gwynne** [comment]
@swift-ci please test

**Comment by gwynne** [comment]
@shahmishal Also added the "slim" variants for AmazonLinux 2 and CentOS 8, and added the CentOS 7 builds (regular and slim both).

**Comment by gwynne** [comment]
@swift-ci please test

**Comment by gwynne** [comment]
@swift-ci test

**Comment by shahmishal** [comment]
Can you split up the 5.2 and 5.3 Dockerfile PR? I am not sure if we will be doing nightly 5.2 toolchain for new Linux platforms at this time.

**Comment by shahmishal** [comment]
Thanks for updating the PR!

**Comment by gwynne** [comment]
@shahmishal Done! All new 5.2 images moved to #198

**Comment by gwynne** [comment]
@swift-ci please test

**Comment by gwynne** [comment]
@swift-ci test

**Comment by gwynne** [comment]
CI failures:
```
20:48:45 [FAILED] - /home/buildnode/jenkins/workspace/swift-docker-PR-linux/swift-ci/master/fedora/rawhide
20:48:45 [FAILED] - /home/buildnode/jenkins/workspace/swift-docker-PR-linux/5.2/ubuntu/18.04/slim
20:48:45 [FAILED] - /home/buildnode/jenkins/workspace/swift-docker-PR-linux/5.2/ubuntu/18.04
20:48:45 [FAILED] - /home/buildnode/jenkins/workspace/swift-docker-PR-linux/5.0/ubuntu/18.04/slim
```
As far as I know there's no action needed from me for these.

**Review by shahmishal — APPROVED** [review]
Thank you!

**Comment by shahmishal** [comment]
```

  | PASSED_nightly-5.3_amazonlinux_2_slim.log
  | PASSED_nightly-5.3_amazonlinux_2.log 
  | PASSED_nightly-5.3_centos_7_slim.log 
  | PASSED_nightly-5.3_centos_7.log 
  | PASSED_nightly-5.3_centos_8_slim.log 
  | PASSED_nightly-5.3_centos_8.log 
  | PASSED_nightly-5.3_ubuntu_16.04_slim.log 
  | PASSED_nightly-5.3_ubuntu_16.04.log 
  | PASSED_nightly-5.3_ubuntu_18.04_slim.log 
  | PASSED_nightly-5.3_ubuntu_18.04.log 
  | PASSED_nightly-5.3_ubuntu_20.04_slim.log 
  | PASSED_nightly-5.3_ubuntu_20.04.log
```

**Pull request merged**